ec2.tf

```diff
@@ -0,0 +1,120 @@
+data "aws_ssm_parameter" "ami" {
+  name = "/aws/service/ecs/optimized-ami/amazon-linux-2/recommended/image_id"
+}
+
+locals {
+  ec2_capacity_providers = local.enabled ? var.capacity_providers_ec2 : {}
+
+  instance_profile_name = join("", aws_iam_instance_profile.default.*.name)
+}
+
+locals {
+  user_data = {
```

Comment:
Allow overriding this entire block as an escape hatch

Reply:
You can override it by specifying `user _data` for ec2 capacity provider

```
locals {
  user_data = {
    for name, provider in local.ec2_capacity_providers :
    name => <<EOT
#!/bin/bash
echo ECS_CLUSTER="${local.cluster_name}" >> /etc/ecs/ecs.config
echo ECS_ENABLE_CONTAINER_METADATA=true >> /etc/ecs/ecs.config
echo ECS_POLL_METRICS=true >> /etc/ecs/ecs.config
echo ECS_ENABLE_SPOT_INSTANCE_DRAINING=${provider.instance_market_options != null && provider.mixed_instances_policy != null} >> /etc/ecs/ecs.config
echo ECS_WARM_POOLS_CHECK=${provider.warm_pool != null} >> /etc/ecs/ecs.config

${replace(provider.user_data, "#!/bin/bash", "")}
EOT
  }
}

```